modules/filesystem/expand.nu

```diff
@@ -0,0 +1,51 @@
+
+# A bash like quick string manipulation script
+# the script works by creating a list from brace contents, seperated by a brace.
+
+# Expand the given string into a list on braces like bashes brace expansion 
+export def main [ 
+  input: string   # the string to expand.
+  ] {
+  listify $input
+}
+
+def listify [lst] {
+  try {
+    ($lst 
+      | parse -r '(?P<left>.*)(?<!\\){(?P<list>.*)(?<!\\)}(?P<right>.*)'
+      | get 0
+      | upsert list { |l|
+        $l.list
+        | split row ","
+        | str trim
+      }
+      | each { |it| 
+        $it.list 
+        | each { |l|
+          listify $"($it.left)($l)($it.right)"
+        }})
+    | flatten
+  } catch {
+    $lst
+  }
+}
+
+export def help [] {
+  print -n ( expand --help )
```

Comment:
This causes an interesting issue on Windows.
```
❯ expand help
Microsoft (R) File Expansion Utility
Copyright (c) Microsoft Corporation. All rights reserved.

Unrecognized switch --.Examples:
  > expand a/{b,c}/d
╭───┬───────╮
│ 0 │ a/b/d │
│ 1 │ a/c/d │
╰───┴───────╯
...
```
It's shelling out and calling Microsoft's expand command.

Reply:
Nope, It does the same thing.
```
❯ expand help

Unrecognized switch -h.Examples:
  > expand a/{b,c}/d
╭───┬───────╮
│ 0 │ a/b/d │
```

Reply:
It's just that it's the name of an external. If I rename it to `exp` instead of `expand`, then it works.
```
❯ exp help
Expand the given string into a list on braces like bashes brace expansion

Usage:
  > main <input>

Flags:
  -h, --help - Display the help message for this command

Parameters:
  input <string>: the string to expand.

Examples:
  > exp a/{b,c}/d
╭───┬───────╮
│ 0 │ a/b/d │
│ 1 │ a/c/d │
╰───┴───────╯
```

Reply:
It is possible windows has problems referencing named function inside of its module. i changed it to `main`. 
It works on my windows laptop, so it should fix this issue.